cmd/config-current.go

```diff
@@ -480,6 +480,12 @@ func lookupConfigs(s config.Config, setDriveCounts []int) {
 	if err != nil {
 		logger.LogIf(ctx, fmt.Errorf("Unable to setup KMS with current KMS config: %w", err))
 	}
+	if kmsCfg.Vault.Enabled {
+		const deprecationWarning = `Native Hashicorp Vault support is deprecated and will be removed on 2021-10-01. Please migrate to KES + Hashicorp Vault: https://github.com/minio/kes/wiki/Hashicorp-Vault-Keystore
+Note that native Hashicorp Vault and KES + Hashicorp Vault are not compatible.
+If you need help to migrate smoothly visit: https://min.io/pricing`
+		logger.LogIf(ctx, fmt.Errorf(deprecationWarning))
+	}
 
```

Comment:
@aead we should also fix our `mc admin info` output currently for KES it says `vault` is online

```
{
  "status": "success",
  "info": {
    "mode": "online",
    "deploymentID": "129c19c9-4cf6-44ff-9f2d-4cb7611be894",
    "buckets": {
      "count": 183
    },
    "objects": {
      "count": 15477
    },
    "usage": {
      "size": 1934131954
    },
    "services": {
      "vault": {
        "status": "online",
        "encrypt": "Ok",
        "decrypt": "Ok"
      },
      "ldap": {}
    },
```

```
~ mc admin info play/ --json
```

Reply:
Yeah 👍 Will send a PR. Do we maintain API stability here or can just change the server response to:
```
"kms": {
        "status": "online",
        "encrypt": "Ok",
        "decrypt": "Ok"
      },
```

Reply:
I think changing this is not a problem @aead it's not directly used by mc for anything specific.